types/fs-extra/index.d.ts

```diff
@@ -170,13 +173,13 @@ export function open(path: string | Buffer, flags: string | number, mode?: numbe
 export function read(fd: number, buffer: Buffer, offset: number, length: number, position: number | null, callback: (err: NodeJS.ErrnoException, bytesRead: number, buffer: Buffer) => void): void;
 export function read(fd: number, buffer: Buffer, offset: number, length: number, position: number | null): Promise<[number, Buffer]>;
 
-export function readFile(filename: string, callback: (err: NodeJS.ErrnoException, data: Buffer) => void): void;
-export function readFile(filename: string, encoding: string, callback: (err: NodeJS.ErrnoException, data: string) => void): void;
-export function readFile(filename: string, options: { flag?: string; } | { encoding: string; flag?: string; }, callback: (err: NodeJS.ErrnoException, data: Buffer) => void): void;
-export function readFile(filename: string, options: { flag?: string; } | { encoding: string; flag?: string; }): Promise<string>;
+export function readFile(filename: string | number, callback: (err: NodeJS.ErrnoException, data: Buffer) => void): void;
```

Comment:
`file` can be `string | Buffer | number` according to the docs: https://nodejs.org/dist/latest-v7.x/docs/api/fs.html#fs_fs_readfile_file_options_callback

Also, let's call this`file` rather than `filename` to indicate the flexibility.